pkg/endpoint/endpoint.go

```diff
@@ -721,6 +721,12 @@ func (e *Endpoint) SetDefaultOpts(opts *option.IntOptions) {
 		e.Options.Opts = option.OptionMap{}
 	}
 
+	// We need to make sure DatapathConfiguration.DisableSipVerification value is same as
+	// the value of SourceIPVerification option in the endpoint.
+	if e.DatapathConfiguration.DisableSipVerification {
```

Comment:
This looks like a code smell to me. This function is supposed to set the Endpoint's default options. All other code in this function looks rather generic, while this if statement is specific to a feature.

Is this code strictly necessary? Shouldn't this make it into `e.Options.Opts` at an earlier stage?

Reply:
should be possible, yeah.. it's not strictly necessary, I only need to make sure this is also correctly set on endpoint restoration

Reply:
fixed, ptal :)
I also added a unit test to make sure this change is covered. 
Btw this was a bit tricky as I found out later that these settings are overriden by calling `SetDefaultConfiguration` in `AddEndpoint()`. I believe the removal of it is safe as all references where this is called (the `AddEndpoint()`) are in the end calling `SetDefaultOpts`.